web/src/views/DandisetLandingView/DandisetDetails.vue

```diff
@@ -152,18 +154,102 @@ export default {
 
       return null;
     },
-    ...mapState('girder', {
-      currentDandiset: (state) => state.currentDandiset,
+    currentVersion() {
+      if (!this.versions) { return draftVersion; }
+      return this.versions[this.currentVersionIndex].version;
+    },
+    currentDandiset() {
+      // Done this way because we'll want to add in
+      // fetching stats from the publish endpoint later on.
+      return this.girderDandiset;
+    },
+    currentVersionIndex() {
+      if (!this.publishDandiset || !this.versions) {
```

Comment:
I like this idea, however even if I do supply this default, there's still a brief moment where it is `null`. If I set the default to `[]`, it goes through these lifecycles:

1. `null`
2. `[]`
3. `[...]`

If this can't be remedied, then it's functionally no better than the previous case. If you know why this might be please lmk. This is what I've done to use defaults:

```
    versions: {
      async get() {
        const { identifier } = this.girderDandiset.meta.dandiset;

        try {
          const { results } = await publishRest.versions(identifier);
          return [
            { version: draftVersion },
            ...results,
          ];
        } catch (err) {
          return [];
        }
      },
      default: [],
    },
```

Reply:
That looks like it should work, according to the docs. If it doesn't work then I agree, it's not worth having.

Reply:
Not sure why, maybe you could see if you could reproduce this locally? In any case, could you link the relevant docs here?

Reply:
Actually it seems like this is moot after b6431f6, since now there's no checks on the truthiness of `versions`.

Reply:
A question for another time then